### PR TITLE
Stat Weights: moved sharable stat calculations to a core module

### PR DESCRIPTION
**Pull request by MartijnHols**
Also updated layout based on Blizzard's character profile:

image

### CODE REVIEW

File: src/Parser/Core/Modules/Features/SpellInfo.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * multiplier: spell scales with whatever procs it, should be ignored for purpose of weights and for 'total healing' number
+ * ignored: spell should be ignored for purpose of stat weights
+ */
+
+// This only works with actual healing events; casts are not recognized.
+export default {
+  [SPELLS.LIGHTS_EMBRACE_HEALING.id]: { // Sea Star of the Depthmother
+    int: false,
+    crit: true,
+    hasteHpct: true, // until LoD's CD is below 8 sec, this speeds up the deck cycle time
```
Comment (kfinch) [inline src/Parser/Core/Modules/Features/SpellInfo.js]: What's LoD?

File: src/Parser/Core/Modules/Features/BaseHealerStatWeights.js [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+import CritEffectBonus from 'Parser/Core/Modules/Helpers/CritEffectBonus';
+import StatTracker from 'Parser/Core/Modules/StatTracker';
+
+import CORE_SPELL_INFO from './SpellInfo';
+import STAT, { getName, getClassNameColor, getIcon } from './STAT';
+
+const DEBUG = true;
+
+// 5% int bonus from wearing all of the highest armor rating (Leather, Mail, Plate) means each new point of int worth 1.05 vs character sheet int
+export const ARMOR_INT_MULTIPLIER = 1.05;
```
[the remark this answers is not in the document]
Comment (MartijnHols) [reply src/Parser/Core/Modules/Features/BaseHealerStatWeights.js]: probably

File: src/Parser/Core/Modules/Features/BaseHealerStatWeights.js [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+    }
+    if (spellInfo.mastery) {
+      this.totalOneMastery += this._mastery(event, healVal);
+    }
+    if (spellInfo.vers) {
+      this.totalOneVers += this._versatility(event, healVal);
+    }
+  }
+  _leech(event, healVal) {
+    const spellId = event.ability.guid;
```
[the remark this answers is not in the document]
Comment (MartijnHols) [reply src/Parser/Core/Modules/Features/BaseHealerStatWeights.js]: good point

File: src/Parser/Core/Modules/Features/BaseHealerStatWeights.js [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+    }
+    const currInt = this.statTracker.currentIntellectRating;
+    // noinspection PointlessArithmeticExpressionJS
+    const healIncreaseFromOneInt = (1 * ARMOR_INT_MULTIPLIER) / currInt;
+    return healVal.effective * healIncreaseFromOneInt;
+  }
+  _getCritChance(event) {
+    const rating = this.statTracker.currentCritRating;
+    const baseCritChance = this.statTracker.baseCritPercentage;
+    const ratingCritChance = rating / this.statTracker.critRatingPerPercent;
```
[the remark this answers is not in the document]
Comment (MartijnHols) [reply src/Parser/Core/Modules/Features/BaseHealerStatWeights.js]: I kinda want to generalize this so we can easily take base crit chance into account everywhere. Take a look at Avenging Wrath, I don't like its current implementation in the statweights, how would we solve it with the stat tracker? probably with another prop

File: src/Parser/Core/Modules/Features/BaseHealerStatWeights.js [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+      const rating = this.statTracker.currentCritRating;
+
+      return effectiveCritHealing * ratingCritChanceContribution / rating;
+    }
+    return 0;
+  }
+  _hasteHpct(event, healVal) {
+    if (healVal.effective === 0) {
+      // While Haste does not directly increase overhealing (or any overhealing in a measurable way), casting a spell that does 0 healing is useless regardless. Being able to cast that one spell more because of the player's Haste gained us no useful output. Note that the (effective) beacon transfer healing caused by this spell will still be added to the Haste value separately.
+      return 0;
```
[the remark this answers is not in the document]
Comment (MartijnHols) [reply src/Parser/Core/Modules/Features/BaseHealerStatWeights.js]: true

File: src/Parser/Core/Modules/Features/BaseHealerStatWeights.js [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+    }
+
+    const healIncreaseFromOneHaste = 1 / this.statTracker.hasteRatingPerPercent;
+
+    return healVal.effective * healIncreaseFromOneHaste;
+  }
+  _mastery(event, healVal) {
+    return 0;
```
[the remark this answers is not in the document]
Comment (MartijnHols) [reply src/Parser/Core/Modules/Features/BaseHealerStatWeights.js]: 👍

File: src/Parser/Core/Modules/Features/BaseHealerStatWeights.js [hunk 7]
```diff
@@ -0,0 +1,355 @@
...
+    const noVersDamage = amount / (1 - currentVersDamageReductionPercentage);
+    return noVersDamage * damageReductionIncreaseFromOneVers;
+  }
+
+  on_toPlayer_heal(event) {
+    this._updateMissingHealth(event);
+  }
+  _updateMissingHealth(event) {
+    if (event.hitPoints && event.maxHitPoints) { // fields not always populated, don't know why
+      this.playerHealthMissing = event.maxHitPoints - event.hitPoints; // TODO: Verify if the `amount` is already included, and leave a comment here stating the results
```
[the remark this answers is not in the document]
Comment (MartijnHols) [reply src/Parser/Core/Modules/Features/BaseHealerStatWeights.js]: I'll update

File: src/Parser/Core/Modules/Features/SpellInfo.js (hunk 1 above)
Comment (MartijnHols) [reply src/Parser/Core/Modules/Features/SpellInfo.js]: Light of Dawn. I don't think we can currently stop it scaling with haste when it passes x%

Comment (kfinch) [reply src/Parser/Core/Modules/Features/SpellInfo.js]: What does Light of Dawn have to do with Sea Star though? Is this like a generalized case that also applies to other spells?

Comment (MartijnHols) [reply src/Parser/Core/Modules/Features/SpellInfo.js]: it procs sea star

Comment (MartijnHols) [reply src/Parser/Core/Modules/Features/SpellInfo.js]: And sea star has an icd

**Comment by kfinch** [comment]
I checked, and apparently Leech does not proc from Absorbs.. I guess we should fix this handling now?

**Comment by MartijnHols** [comment]
In another PR

**Comment by MartijnHols** [comment]
Thanks for the review!

**Pull request merged**